TC-02: Setup datastore with redux

Haris-moin opened the pull request:
- setup redux store
- create redux structure
- implement redux actions , reducer and state
- test redux actions and state in example component

hmoin-nisum-com committed:
TC-01: Setup datastore with redux

- setup redux store
- create redux structure
- implement redux actions , reducer and state
- test redux actions and state in example copmonent

hiqbalmallick commented on src/store/modules/example/reducer.ts:
```diff
@@ -0,0 +1,16 @@
+export const exampleReducer = (state = 0, action: Action) => {
+  switch (action.type) {
+    case EXAMPLE_REQUEST_TYPES.EXAMPLE_INCREAMENT_COUNTER:
+      return state + 1;
```
This is not the right way to update the state in the reducer.

Haris-moin replied:
resolved it in correct way

hiqbalmallick commented on src/example/counter/index.tsx:
```diff
@@ -0,0 +1,34 @@
+  return (
+    <div>
+      <button
+        onClick={() => {
+          dispatch(increamentCouter());
+        }}
```
create a method above and pass the reference here

Haris-moin replied:
resolved it by passing a reference

hiqbalmallick commented on src/example/counter/index.tsx:
```diff
@@ -0,0 +1,34 @@
+      </button>
+      <h2>Count: {count}</h2>
+      <button
+        onClick={() => {
+          dispatch(decreamentCouter());
+        }}
```
pass method reference here

Haris-moin replied:
resolved it by passing a reference

hmoin-nisum-com committed:
TC-02: Setup datastore with redux

-changed state into correct formate and manipulate state according to action
-remove callback from tags event and passed a method asa reference
